src/main/java/org/jitsi/jicofo/jigasi/TranscriberManager.java

```diff
@@ -0,0 +1,312 @@
+     * Transcriber from the smack thread updating presence.
+     */
+    private ExecutorService executorService;
+
+    /**
+     * Create a {@link TranscriberManager} responsible for inviting Jigasi as
+     * a transcriber when this is desired.
+     *
+     * @param protocolProviderHandler the handler giving access a XmppConnection
+     * @param chatRoom the room of the conference being managed
+     * @param jigasiDetector detector for Jigasi instances which can be dialed
+     * to invite a transcriber
+     */
+    public TranscriberManager(ProtocolProviderHandler protocolProviderHandler,
+                              ChatRoom chatRoom,
+                              JigasiDetector jigasiDetector)
+    {
+        this.connection
+            = protocolProviderHandler.getOperationSet(
+            OperationSetDirectSmackXmpp.class).getXmppConnection();
+
+        this.chatRoom = chatRoom;
+        this.jigasiDetector = jigasiDetector;
+    }
+
+    /**
+     * Initialise the manager by starting to listen for
+     * {@link ChatRoomMemberPropertyChangeEvent}s.
+     */
+    public void init()
+    {
+        if(executorService != null)
+        {
+            executorService.shutdown();
+            executorService = null;
+        }
+
+        executorService = Executors.newSingleThreadExecutor();
+        chatRoom.addMemberPropertyChangeListener(this);
+
+        logger.debug("initialised transcriber manager");
+    }
+
+    /**
+     * Stop the manager by stopping to listen for
+     * {@link ChatRoomMemberPropertyChangeEvent}s.
+     */
+    public void dispose()
+    {
+        executorService.shutdown();
+        executorService = null;
+        chatRoom.removeMemberPropertyChangeListener(this);
+
+        logger.debug("disposed transcriber manager");
+    }
+
+    /**
+     * Listener method for each {@link ChatRoomMemberPropertyChangeEvent}, which
+     * can possibly indicate the transcriber needs to be invited.
+     *
+     * {@inheritDoc}
+     */
+    @Override
+    public void chatRoomPropertyChanged(ChatRoomMemberPropertyChangeEvent event)
+    {
+        if(event.getNewValue() instanceof Presence)
+        {
+            onNewPresence(event);
+        }
+    }
+
+    /**
+     * Check whether a {@link ChatRoomPropertyChangeEvent} is due to a new
+     * {@link Presence}, and when it is, deal with the information is has
+     * provided.
+     *
+     * @param event the event to check the {@link Presence} off
+     */
+    private void onNewPresence(ChatRoomMemberPropertyChangeEvent event)
+    {
+        Presence presence = getPresenceOrNull(event);
+
+        if(presence == null)
+        {
+            return;
+        }
+
+        if(containsTranscriptionStatus(presence))
+        {
+            maybeStoppedTranscribing(presence);
+        }
+        if(isRequestingTranscriber(presence) && !active)
+        {
+            executorService.submit(this::startTranscribing);
```

Comment:
We only update the 'active' flag after we get the response from jigasi (over the network). If we receive a second request (e.g. someone clicking the button rapidly) we'll submit two tasks and invite two jigasi's. What do you think about updating 'active' here, before submitting the task (the task can then set it to 'false' if something fails)?